model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/common/HotRodTypesUtils.java

```diff
@@ -38,7 +39,7 @@
     }
 
     public static <MapKey, MapValue, SetValue> Map<MapKey, MapValue> migrateSetToMap(Set<SetValue> set, Function<SetValue, MapKey> keyProducer, Function<SetValue, MapValue> valueProducer) {
-        return set == null ? null : set.stream().collect(Collectors.toMap(keyProducer, valueProducer));
+        return set == null ? null : set.stream().collect(HashMap::new, (m, v) -> m.put(keyProducer.apply(v), valueProducer.apply(v)), HashMap::putAll);
```

Comment:
This change is required because otherwise it threw NPE if `HotRodPair`'s value is `null`. 
```
Caused by: java.lang.NullPointerException
at java.base/java.util.Objects.requireNonNull(Objects.java:221)
at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:178)
at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1621)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
at org.keycloak.models.map.storage.hotRod.common.HotRodTypesUtils.migrateSetToMap(HotRodTypesUtils.java:41)
at org.keycloak.models.map.storage.hotRod.client.HotRodProtocolMapperEntityDelegate.getConfig(HotRodProtocolMapperEntityDelegate.java:70)
at org.keycloak.models.map.client.MapProtocolMapperEntityCloner.deepClone(MapProtocolMapperEntityCloner.java:6)
```

Reply:
Would it be sufficient to use something like this:
```suggestion
        return set == null ? null : set.stream().collect(Collectors.toMap(keyProducer, valueProducer, (a, b) -> a));
```

Reply:
There is NPE with this solution.

Reply:
Not for me. In my case, the third parameter makes it not checking the value for `null`. Anyway, this probably depends on Java implementation and it seems to be a [known bug](https://bugs.openjdk.java.net/browse/JDK-8148463) in OpenJDK. 

I would say, it is safer to use your approach even though the lambda is not ideal.